WireDomain/Sources/WireDomain/Synchronization/PullAllFeatureConfigsSync.swift

```diff
@@ -0,0 +1,127 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import WireAPI
+
+// sourcery: AutoMockable
+protocol PullAllFeatureConfigsSyncProtocol {
+
+    func pull() async throws
+
+}
+
+/// An object to keep the local feature configs up to date
+/// with the remote feature configs.
+
+struct PullAllFeatureConfigsSync: PullAllFeatureConfigsSyncProtocol {
+
+    private let api: any FeatureConfigsAPI
+    private let store: any FeatureConfigLocalStoreProtocol
+
+    init(
+        api: any FeatureConfigsAPI,
+        store: any FeatureConfigLocalStoreProtocol
+    ) {
+        self.api = api
+        self.store = store
+    }
+
+    /// Fetch all conversation labels from remote, then create and update
+    /// them locally.
```

Comment:
Also this description I would probably place at the protocol level.

The caller won't see it otherwise.